apps/server-asset-sg/src/features/files/file-s3.service.ts

```diff
@@ -78,14 +71,22 @@ export class FileS3Service {
         new DeleteObjectCommand({
           Key: this.getKey(name),
           Bucket: this.bucket,
-        })
+        }),
       );
       return true;
     } catch (e) {
       return handleS3Error(e) ?? false;
     }
   }
 
+  private createMetadata(output: HeadObjectCommandOutput, name: string): FileS3Metadata {
+    return {
+      name,
+      byteCount: output.ContentLength ?? null,
+      mediaType: output.ContentType ?? null,
+      pageCount: Number(output.Metadata?.pagecount) ?? null,
```

Comment:
What is the expected behavior with this null coalesce here? If it is meant to be set to `null` when the `pagecount` doesn't exist, then it doesn't work right now, as `Number` returns `NaN` for any non-number, which doesn't coalesce.
```suggestion
      pageCount: output.Metadata?.pagecount ? Number(output.Metadata.pagecount) : null,
```

Reply:
Good catch, that was my intention :)